apps/emqx_management/src/emqx_mgmt_api_clients.erl

```diff
@@ -1197,6 +1331,79 @@ format_persistent_session_info(ClientId, PSInfo0) ->
     ),
     result_format_undefined_to_null(PSInfo).
 
+format_msgs_resp(MsgType, Msgs, Meta, QString) ->
+    #{
+        <<"payload">> := PayloadFmt,
+        <<"max_payload_bytes">> := MaxBytes
+    } = QString,
+    Meta1 = emqx_mgmt_api:encode_after_pager_params(Meta, after_encoding(MsgType)),
+    Resp = #{meta => Meta1, data => format_msgs(Msgs, PayloadFmt, MaxBytes)},
+    %% Make sure minirest won't set another content-type for self-encoded JSON response body
+    Headers = #{<<"content-type">> => <<"application/json">>},
+    case emqx_utils_json:safe_encode(Resp) of
```

Comment:
Note: this can actually serialize pretty much every payload, as it is not parsed and validated. 
For example, if message.payload is invalid JSON string (missing enclosing `}`), it won't be detected:
````
emqx_utils_json:safe_encode(#{message => <<"{\"foo\": \"bar\"">>}).
{ok,<<"{\"message\":\"{\\\"foo\\\": \\\"bar\\\"\"}">>}
```